public/editor/scripts/project/project.js

```diff
@@ -246,12 +247,33 @@ define(function(require) {
 
             // Look for an HTML file to open, ideally index.html
             var indexPos = 0;
+            var foundIndexHTML = false;
             found.forEach(function(path, idx) {
               if(Path.basename(path) === "index.html") {
                 indexPos = idx;
+                foundIndexHTML = true;
               }
             });
 
+            // Create a default index.html file
+            if (!foundIndexHTML) {
+              var location = "/default-files/html.txt";
+              $.get(location).then(function(data) {
+                var file = getRoot() + "/index.html";
+                _fs.writeFile(file, data, function(err) {
+                  if (err) {
+                    return console.error("Cannot write file to project: " + err);
+                  }
```

Comment:
It's a bit more subtle than that--we do need it for sure, but we also need to call it in two spots, and we have to pass along the path to the file to open.  The block of code should look something like this (I haven't tested this code, but it's the basic idea you want):

```js
            if(foundIndexHTML) {
              callback(null, found[indexPos]);
              return;
            }

            // Create a default index.html file since we didn't find any
            var location = "/default-files/html.txt";
            $.get(location).then(function(data) {
              var file = getRoot() + "/index.html";
              _fs.writeFile(file, data, function(err) {
                if (err) {
                  console.error("Cannot write file to project: " + err);
                  return callback(err);
                }
                callback(null, file);
              });
            }, function(err) {
              if (err) {
                console.error("Failed to download " + location, err);
                callback(err);
                return;
              }
            });
```